I18n release

danimoh opened the pull request.

danimoh committed:
Export: style fixes

- remove close button background color
- fix login file download button line-height
- reduce recovery words intro warning margin
- limit max width of continue button on recovery words intro
- adapt margins in login file intro
- keep displaying login-file-success on page flip to recovery words
- set restrict height of blurred background pages to default page height
- fix opacity and hover of back button on recovery words page
- center text in login-file-success also if it breaks into multiple lines

danimoh committed:
package.json: update @nimiq/style

Update for fix of text color in inverse buttons on dark background

danimoh committed:
ImportWords,ChangePassword: hide download skip button on download start

danimoh committed:
FileImporter: correctly center file import translations

danimoh committed:
DownloadLoginFile: reduce image size when there's not enough space

danimoh committed:
FileImporter: internationalize error messages

danimoh committed:
RemoveKey: center label confirmation also for longer text

danimoh committed:
ChangePassword: avoid increasing card size for long translations

danimoh committed:
Internationalization: update translations; add Spanish

danimoh commented:
I repurposed this PR to a general PR for "I18n release" changes similar to the one in the Hub.
@sisou please re-review the newest changes.

Additional changes might be added later depending on feedback

sisou commented on src/translations/de.json:
```diff
@@ -22,21 +22,23 @@
 
     "import-heading-enter-recovery-words": "Wiederherstellungswörter eingeben",
-    "import-import-login-file": "Lade dein Login-File hoch",
+    "import-import-login-file": "Importiere mittels Login‑Datei",
```
I think the wording should be about "login", not "import". So rather "Log dich mittels Login-Datei ein"

danimoh committed:
common.css: avoid global-close button causing scroll on mobile

danimoh committed:
TopLevelApi: support app specific global-close internationalization

danimoh committed:
RemoveKey: allow freely positioning {accountLabel} in translation

For some languages like Korean the order of the sentence is different than in
English.

danimoh committed:
Internationalization: update translations

danimoh committed:
Internationalization: restrict initial release to en and zh

danimoh committed:
Internationalization: use browser lang if no lang cookie set

danimoh commented:
This should now be all changes for the release.
Please re-review.

mraveux commented on src/translations/fr.json:
```diff
@@ -3,7 +3,15 @@
     "language-changed": "La langue d'affichage a changé. Voulez-vous recharger la page pour mettre à jours toutes les traductions ? Sinon certaines pourraient ne pas être mises à jour automatiquement.",
 
-    "back-to": "Retour à ",
+    "back-to-app": "Back to {appName}",
```
This one and the others below are now translated in transifex.

mraveux commented on src/translations/fr.json:
```diff
@@ -158,8 +166,7 @@
     "remove-key-show-recovery-words": "Créer une sauvegarde",
-    "remove-key-enter-label-1": "Écrivez",
-    "remove-key-enter-label-2": "pour vous déconnecter.",
+    "remove-key-label-confirm-instructions": "Type \"{accountLabel}\" to log out.",
```
Translated too

mraveux reviewed (approved):
Just some missing French translations, otherwise it's all good.

sisou commented on src/request/TopLevelApi.js:
```diff
@@ -229,15 +229,60 @@ class TopLevelApi extends RequestParser {
-        $globalCloseText.textContent = buttonText;
+
+        // eslint-disable-next-line require-jsdoc-except/require-jsdoc
+        const setButtonText = () => {
```
```suggestion
        function setButtonText() {
```

danimoh replied:
In my opinion the way it is currently is the nicer way of defining a function within a function.

sisou commented on src/request/remove-key/RemoveKey.js:
```diff
@@ -58,7 +59,18 @@ class RemoveKey {
         }
 
-        $labelSpan.textContent = this._request.keyLabel;
+        // eslint-disable-next-line require-jsdoc-except/require-jsdoc
+        const setLabelConfirmInstructions = () => {
```
```suggestion
        function setLabelConfirmInstructions() {
```

sisou commented on src/translations/de.json:
```diff
@@ -3,15 +3,23 @@
+    "back-to-wallet": "Zurück zur {Wallet}",
+    "back-to-miner": "Zurück zum {Nimiq Miner}",
+    "back-to-faucet": "Zurück zur {Nimiq Faucet}",
+    "back-to-donation": "Zurück zur Spende",
```
@danimoh I believe this is not a good translation. This is the _Donation Button Maker app_, not a donation itself.

danimoh replied:
The app itself is called `Nimiq Donation`, thus this is a translation close to the original `Back to Nimiq Donation`.
However, to resolve this, I the donation app now identifies itself as `Donation Button Creator` and I adapted the translations accordingly.

sisou commented on src/translations/de.json:
```diff
@@ -3,15 +3,23 @@
+    "back-to-donation": "Zurück zur Spende",
+    "back-to-gift-card": "Zurück zur Geschenkkarte",
+    "back-to-vote": "Zurück zur Abstimmung",
+    "back-to-cpl": "Zurück zu CryptoPayment.link",
```
@danimoh I think this is too long, this would be shorter:
```suggestion
    "back-to-cpl": "Zurück zum Link",
```

danimoh replied:
That's too unspecific imo and `Zurück zu CryptoPayment.link` still fits on mobile.

danimoh committed:
RemoveKey: normalize soft hyphens in labels
